### PR TITLE
build: replace Lerna with a custom script

### PR DESCRIPTION
**Pull request by GermainBergeron**
### Proposed Changes

Replaced Lerna with a script that uses https://github.com/coveo/semantic-monorepo-tools. This will give us more flexibility in the long run!

I have a PR in the other repo that must be deployed first

### Potential Breaking Changes

We'll have to see on master if the publish happens correctly.

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: build/publishNewVersion.mjs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    .option('--tag', 'tag to use on NPM', 'latest')
+    .option('--branch', 'allow deploy on branch', 'master')
+    .option('--force <packages...>', 'force a bump on specified package', ['root'])
+    .addOption(new Option('--bump <type>', 'bump a <type> version instead of reliying on commit messages').choices(BUMP_TYPES))
+
+program.parse();
+
+const options = program.opts();
+
+// We consider that root always change, we always want to bump a version
```
Comment (gdostie) [inline build/publishNewVersion.mjs:37]:
```suggestion
// We consider that root always changes, we always want to bump a version
```

File: Jenkinsfile.groovy [hunk 2]
```diff
@@ -202,54 +202,24 @@ pipeline {
...
-            --conventional-prerelease \
-            --preid next \
-            --dist-tag next \
-            --no-commit-hooks \
-            --force-publish \
-            --no-push \
-            --no-git-reset \
-            --yes"
+            sh "node --experimental-specifier-resolution=node build/publishNewVersion.mjs \
+            --bump release \
```
Comment (aiheon) [inline Jenkinsfile.groovy:211]: do you think this would handle well merging back Next into master? :thinking:

File: build/publishNewVersion.mjs [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                    gitPush();
+                    gitPushTags();
+
+                    const forcePackagesWithoutRoot = options.force.filter(packageName => packageName !== 'root');
+                    pnpmPublish(since, options.tag, options.branch, forcePackagesWithoutRoot, ['root']);
+                }
+            }
+
+            // const octokit = new Octokit({auth: process.env.GITHUB_CREDENTIALS});
+            // await createGitHubRelease(
```
Comment (aiheon) [inline build/publishNewVersion.mjs:106]: WIP?

File: package.json [hunk 4]
```diff
@@ -35,7 +38,7 @@
     },
     "husky": {
         "hooks": {
-            "pre-commit": "lerna run --concurrency 1 --stream precommit",
+            "pre-commit": "ultra --recursive --serial --root precommit",
```
Comment (aiheon) [inline package.json:41]: :love_hotel:

File: build/publishNewVersion.mjs (hunk 3 above)
Comment (GermainBergeron) [reply build/publishNewVersion.mjs:106]: Yes, I'll make sure the rest work before trying to create a release (we don't have those anymore)

File: Jenkinsfile.groovy (hunk 2 above)
Comment (GermainBergeron) [reply Jenkinsfile.groovy:211]: I'm guessing that we'll have to do a commit in the PR that merges into master to bump a major

**Review by aiheon — APPROVED** [review]

**Pull request merged**